diff --git a/node_modules/solid-start-mdx/index.js b/node_modules/solid-start-mdx/index.js
index a11f749..e13529e 100644
--- a/node_modules/solid-start-mdx/index.js
+++ b/node_modules/solid-start-mdx/index.js
@@ -77,7 +77,7 @@ function jsToTreeNode(jsString, acornOpts) {
   };
 }
 
-export default async function () {
+export default async function (config) {
   const cache = new Map();
   const headingsCache = new Map();
   const frontMatterCache = new Map();
@@ -190,8 +190,7 @@ export default async function () {
           {
             disableImplicitReactImport: true,
             includeJSDocInHover: true,
-            // theme: "css-variables",
-            themes: ["github-dark", "github-light"],
+            themes: config.themes || ["github-dark", "github-light"],
             defaultCompilerOptions: {
               allowSyntheticDefaultImports: true,
               esModuleInterop: true,
